src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -741,8 +743,24 @@ private String formatDate(final Date date) {
 		try {
 			return TransactionDto.class.getConstructor(Integer.class, String.class, String.class,  Double.class,
 			 String.class, Date.class, Integer.class, Integer.class, String.class, String.class, Integer.class, String.class, String.class);
-		} catch (NoSuchMethodException ex) {
+		} catch (final NoSuchMethodException ex) {
 			throw new RuntimeException(ex);
 		}
 	}
+
+
+	public Transaction saveTransaction(final TransactionDto transactionDto, final Lot lot) {
```

Comment:
@vmaletta why the entire lot is needed? Does the transaction contain the lotId? In that case the lot param is not needed. Why the status is not a param?